sim/include/index/entityIndex.h

```diff
@@ -27,27 +29,31 @@ class EntityIndex {
         };
     }
 
-    EntityId addEntityId() {
-        // Update the next entity id
-        auto entityId = nextEntityId;
-        nextEntityId++;
-
-        // Update the entityCompMap
-        entityCompMap.emplace(entityId, std::unordered_set<CompStoreId>());
-        return entityId;
+    EntityId addEntityId();
+
+    // Start with certain entity IDs
+    template <class T>
+    // Require that the input is an iterable, and the value stored in the
+    // iterable is an EntityId
+    requires std::ranges::range<T>&&
+        std::is_same_v<std::ranges::range_value_t<T>, EntityId> void
+        setEntityIds(T ids) {
```

Comment:
Why must this function be templated instead of accepting a vector/list of `EntityId`?

Reply:
If I specify `std::vector` or `std::list`, they will only allow either one, not the other. I am trying to generalise so that the function works for any array-type that is iterable

Reply:
:disappointed: Why does c++ have no alterable interface. This is so much more larder to read